src/mono/wasm/runtime/blazor/WebAssemblyConfigLoader.ts

```diff
@@ -0,0 +1,50 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+import { Module } from "../imports";
+import { BootConfigResult } from "./BootConfig";
+import { WebAssemblyStartOptions } from "./WebAssemblyStartOptions";
+
+export class WebAssemblyConfigLoader {
+    static async initAsync(bootConfigResult: BootConfigResult, startOptions: Partial<WebAssemblyStartOptions>): Promise<void> {
+        // TODO MF: Hook getApplicationEnvironment
+        // Blazor._internal.getApplicationEnvironment = () => bootConfigResult.applicationEnvironment;
+
+        const configFiles = await Promise.all((bootConfigResult.bootConfig.config || [])
+            .filter(name => name === "appsettings.json" || name === `appsettings.${bootConfigResult.applicationEnvironment}.json`)
+            .map(async name => ({ name, content: await getConfigBytes(name) })));
+
+        const anyConfig = Module.config as any;
+        if (anyConfig.onAppConfigLoaded) {
+            anyConfig.onAppConfigLoaded(configFiles);
+        }
+
+        // TODO MF: Hook getConfig
+        // Blazor._internal.getConfig = (fileName: string) : Uint8Array | undefined => {
```

Comment:
I think we want to include this logic as Icall in the `System.Runtime.InteropServices.JavaScript` not as hook back to blazor.

Reply:
Do you think that it is possible to do it separately? Or is it that simple that we should do it right away?